vortex-array/src/builders/primitive.rs

```diff
@@ -45,6 +46,55 @@ impl<T: NativePType> PrimitiveBuilder<T> {
             None => self.append_null(),
         }
     }
+
+    pub fn patch(&mut self, patches: Patches, starting_at: usize) -> VortexResult<()> {
+        let (array_len, offset, indices, values) = patches.into_parts();
+        assert!(starting_at + array_len == self.len());
+        let indices = indices.into_primitive()?;
+        let values = values.into_primitive()?;
+        let validity = values.validity_mask()?;
+        let values = values.as_slice::<T>();
+        match_each_unsigned_integer_ptype!(indices.ptype(), |$P| {
+            for (compressed_index, decompressed_index) in indices.into_primitive()?.as_slice::<$P>().into_iter().enumerate() {
+                let decompressed_index = *decompressed_index as usize;
+                let out_index = starting_at + decompressed_index - offset;
+                if validity.value(compressed_index) {
+                    self.values[out_index] = values[compressed_index]
+                } else {
+                    self.nulls.set_bit(out_index, false)
+                }
```

Comment:
```suggestion
                self.values[out_index] = values[compressed_index];
                self.nulls.set_bit(out_index, validity.value(compressed_index));
```
I think you can remove the branch if you ensure that `self.nulls` has the correct backing array below

Reply:
I did something different but indeed there are no branches.

Reply:
I did this.